### ISSUE TITLE
How to set limits on flow data without affecting geom_gate

### ISSUE THREAD

**Issue by mikejiang** [RGLab/ggcyto#5]
Flow data often time has off-boundary events

``` r
p <- ggcyto(gs, aes(x = CD4, y = CD8), subset = "CD3+") 
p <- p + geom_hex(bins = 64)
p <- p + geom_gate("CD8")
p
```

![image](https://cloud.githubusercontent.com/assets/1385649/11671917/80a8cfd0-9dc0-11e5-87df-c3720e105ef4.png)

And we've been trying to remove them by ggplot's native `xlim/ylim` (i.e. `Scale`)  layer.

``` r
 p + xlim(c(0,4e3))
```

![image](https://cloud.githubusercontent.com/assets/1385649/11672035/b8295578-9dc1-11e5-8ef9-8bd089e42665.png)

The side effect of `scale` transformation also causes the data in `geom_gate` layer (especially `polygonGate`)  to be truncated and thus will lose its geometrical shape. 

There is some discussion about having `ggplot` to support layers that are free from `scale` limits 
https://github.com/hadley/ggplot2/issues/187
However no actions have been taken by @hadley .

@raphg suggested the `interpolation` remedy a while back. The idea was to ensure `polygon` has sufficient number of points so that it still keeps its shape even after losing some vertices.

This was working fine until we encountered the extended gates (from `flowJo` or `gatingML`) that has exaggerated vertices on the far left, which makes it difficult to interpolate enough points within the displayed range that can prevent the polygon from losing shape.

As discussed before, we can't use `coord_trans` here either because we do need to remove these off-bound data through `scale` transformation because large number of these events often screw up the `scale_fill_gradientn` for `hexbin`.

**Comment by mikejiang** [comment]
@gfinak , looking at [gate extension code](https://github.com/RGLab/flowWorkspace/blob/trunk/src/gate.cpp#L196-L210) for parsing `flowJo` xml, it is already extending the gates to the actual data limits. But this step is served as the `preprocessing` to gates prior to any actual `gating`, thus only `raw`(root node) data is available for lookup.  Therefore, the extended points can still be far left.  It is not impossible to modify the code to move the `gate extension` into the `gating` step. 

The same issue exists for `parse.gatingML`. 

So I still wonder if there is something cheaper we can do on `ggcyto` side.

**Comment by mikejiang** [comment]
The `polygonGate` vertices are truncated by the actual `data range` compared to the original `vertices` points. 
![image](https://cloud.githubusercontent.com/assets/1385649/23383896/612d8c16-fcfd-11e6-9d29-6ff7222c1e69.png)

This truncation was originally for preventing interpolation on the very extreme polygon vertices (e.g. `Inf` generated by `openCyto`). So we should only skip it by default.

**Comment by mikejiang** [comment]
https://support.bioconductor.org/p/107606/#107657,  it should have been taken care of by `polygon interpolation`

**Comment by mikejiang** [comment]
fixed by bc182dae8f177ba75900b0ae420e600d51a12425

**Comment by mikejiang** [comment]
addressed by 6f0fa93ddbbc12227015aa3804c3c7fa4a4ac229
